lib/knapsack_pro/repository_adapters/git_adapter.rb

```diff
@@ -14,8 +14,34 @@ def branches
         str_branches.split("\n")
       end
 
+      def authors
+        git_authors
+          .split("\n")
+          .map { |line| line.strip }
+          .map { |line| line.split("\t") }
+          .map do |commits, author|
+            { commits: commits.to_i, author: KnapsackPro::MaskString.call(author) }
+          end
+      rescue Exception
```

Comment:
I thought about just `rescue`ing the git command but I wanted to go 100% safe

Reply:
Ok.

I tested using a fake command `xgit`:

```
3.2.2 :005 >  def git_authors
3.2.2 :006 >         `xgit shortlog --summary --email --since "one month ago"`
3.2.2 :007 >  end


3.2.2 :008 > git_authors
sh: xgit: command not found
 => ""

3.2.2 :015 > git_authors
3.2.2 :016 >           .split("\n")
3.2.2 :017 >           .map { |line| line.strip }
3.2.2 :018 >           .map { |line| line.split("\t") }
3.2.2 :019 >           .map do |commits, author|
3.2.2 :020?>             { commits: commits.to_i, author: KnapsackPro::MaskString.call(author)
}
3.2.2 :021 >           end
sh: xgit: command not found
 => []
```

Is it fine to return an empty array here? Maybe it would be better to return:

```
[{ commits: 0, author: "no git <no.git@example.com>" }]
```

Reply:
I'm not sure I follow. The entire body of `authors` is rescued, so that would always return `[{ commits: 0, author: "no git <no.git@example.com>" }]` in case of an exception.

Also, `git_authors` is not public and it's only called by `authors`.

Reply:
I'm not sure I follow.

The entire body of authors is rescued, so that would always return [{ commits: 0, author: "no git <no.git@example.com>" }] in case of an exception.

Also, git_authors is not public and it's only called by authors.

Example:

```bash
➜ irb
3.2.2 :001 > def cmd; `xgit`; end
 => :cmd 
3.2.2 :002 > def go
3.2.2 :003 >   cmd
3.2.2 :004 > rescue Exception
3.2.2 :005 >   "rescued"
3.2.2 :006 > end
 => :go 
3.2.2 :007 > go
 => "rescued" 

```

Reply:
Hmm. When I use the above example with the `cmd` method I also get the exception but when I use this:

```
def git_authors
  `xgit shortlog --summary --email --since "one month ago"`
end
```

then there is no exception raised. Only empty string is returned and the `sh: xgit: command not found` message is printed to the output.

```
3.2.2 :008 > git_authors
sh: xgit: command not found
 => ""
```

My main point is that we should also cover case when the method returns an empty string because catching exception seems to not be enough.